director/origin_api_test.go

```diff
@@ -188,3 +193,182 @@ func TestNamespaceKeysCacheEviction(t *testing.T) {
 		}
 	})
 }
+
+// TestNamespaceKeysCacheTTLExpiration tests that the namespaceKeys cache
+// properly expires keys after the TTL period and that WithDisableTouchOnHit
+// prevents TTL refresh on cache access. This validates the fix for the bug
+// where stale keys would remain cached indefinitely if accessed frequently.
+func TestNamespaceKeysCacheTTLExpiration(t *testing.T) {
+	ctx, cancel, egrp := test_utils.TestContext(context.Background(), t)
+	defer func() { require.NoError(t, egrp.Wait()) }()
+	defer cancel()
+
+	server_utils.ResetTestState()
+
+	// Use a shorter TTL for testing (2 seconds instead of 15 minutes)
+	originalTTL := param.Director_AdvertisementTTL.GetDuration()
+	defer viper.Set(param.Director_AdvertisementTTL.GetName(), originalTTL)
```

Comment:
You shouldn't need to defer re-setting viper variables -- as a matter of principle, all viper config should be cleaned up by the test itself.